### PR TITLE
Misc Fixes to Plugin

### PR DESCRIPTION
**Pull request by RadWolfie**
Two fixes plugin require for Windows platform to work correctly are:
 * Fix buffer's stream being full and unable to flush into output buffer reader. Without this fix, Ghidra Xbe plugin will cause deadlock for XbSymbolDatabaseTool executable on Windows platform. (Not verified on other platforms.)
 * Ghidra 9.1.x has a bug for GUI version causing extra slash before drive letter when it's not supposed to. Yet somehow it doesn't affect opening xbe files. Headless version of analyzer does not have this problem.
   * This hack check if there is single forward slash before drive letter will enforce standard Windows' file path before attempt to use XbSymbolDatabaseTool exectuable. Although, it also check if 2nd character is not forward slash which is known for network drive usage.
   * I am not sure, it's possible Ghidra 2.x may have resolve the issue. I had not verified this and currently isn't compatible with Ghidra 2.x straight forward.

### CODE REVIEW

File: src/main/java/XbeLoader/XbeXbSymbolDatabaseAnalyzer.java [hunk 1]
```diff
@@ -92,14 +92,20 @@ public boolean added(Program program, AddressSetView set, TaskMonitor monitor, M
 			return false;
 		}
 		String xbePath = program.getExecutablePath();
+		// HACK: Somehow GUI broke this yet headlessAnalyzer didn't...
+		// Even if remove first forward slash before drive letter cause corruption to "Executable Location" meta, not even sure why either.
+		// Plus extra check if using network drive for safety precaution.
+		if (Platform.CURRENT_PLATFORM.getOperatingSystem() == OperatingSystem.WINDOWS && xbePath.toCharArray()[0] == '/' && xbePath.toCharArray()[1] != '/') {
```
Comment (mborgerson) [inline src/main/java/XbeLoader/XbeXbSymbolDatabaseAnalyzer.java:98]: nits:
* Could you please shorten/wrap this line please
* Might change `xbePath.toCharArray()[0] == '/'` to `(xbePath.charAt(0) == '/')`

File: src/main/java/XbeLoader/XbeXbSymbolDatabaseAnalyzer.java [hunk 2]
```diff
@@ -92,14 +92,20 @@ public boolean added(Program program, AddressSetView set, TaskMonitor monitor, M
 			return false;
 		}
 		String xbePath = program.getExecutablePath();
+		// HACK: Somehow GUI broke this yet headlessAnalyzer didn't...
+		// Even if remove first forward slash before drive letter cause corruption to "Executable Location" meta, not even sure why either.
```
Comment (mborgerson) [inline src/main/java/XbeLoader/XbeXbSymbolDatabaseAnalyzer.java:96]: This is strange...

nits: Please change comment to something shorter, e.g.

```
// Ensure path does not erroneously begin with `/` before drive letter
```

File: src/main/java/XbeLoader/XbeXbSymbolDatabaseAnalyzer.java [hunk 3]
```diff
@@ -111,6 +117,8 @@ public boolean added(Program program, AddressSetView set, TaskMonitor monitor, M
 				String name = fullName.substring(libNameLength+2);
 				program.getSymbolTable().createLabel(address, name, getNamespace(program, lib), SourceType.ANALYSIS);
 			}
+			
+			exec.waitFor();
```
Comment (mborgerson) [inline src/main/java/XbeLoader/XbeXbSymbolDatabaseAnalyzer.java:121]: Makes sense

**Comment by mborgerson** [comment]
Thanks. Some minor feedback items left in review. I'll merge once fixed

File: src/main/java/XbeLoader/XbeXbSymbolDatabaseAnalyzer.java (hunk 1 above)
Comment (RadWolfie) [reply src/main/java/XbeLoader/XbeXbSymbolDatabaseAnalyzer.java]: Done, it's been a long time I had touched java coding. Yet still not an expert with it.

File: src/main/java/XbeLoader/XbeXbSymbolDatabaseAnalyzer.java (hunk 2 above)
Comment (RadWolfie) [reply src/main/java/XbeLoader/XbeXbSymbolDatabaseAnalyzer.java]: Done, plus extend a little bit for origin of cause.

**Comment by mborgerson** [comment]
I've merged your patches but amended the commit messages to be more descriptive. I did not have permission to update your branch so I've just pushed to my own master and will close this PR now. Thanks for the patches!

**Comment by RadWolfie** [comment]
Could had make another review to fix it. Except it's your repository, so whatever.

**Pull request closed without merging**